internal/advertisement-operator/controller.go

```diff
@@ -127,6 +134,23 @@ func (r *AdvertisementReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
 			return ctrl.Result{}, err
 		}
 		klog.Info("Correct creation of virtual kubelet deployment for cluster " + adv.Spec.ClusterId)
+		// start the keepalive check for the new cluster
+		r.checkRemoteCluster[adv.Spec.ClusterId] = new(sync.Once)
+		go r.checkRemoteCluster[adv.Spec.ClusterId].Do(func() {
+			err := r.checkClusterStatus(adv)
+			// if everything works well, the check is an infinite loop
+			// therefore, if an error is returned, the foreign cluster is not reachable anymore
+			if err != nil {
+				// the foreign cluster is down
+				klog.Error(err)
+				//TODO: decide which action to do (delete the adv/peering request...
```

Comment:
I suggest to have two different behaviors, similarly as done by the kube-controller manager:
-  ```--node-monitor-grace-period duration```    Default: 40s  Amount  of time which we allow running Node to be unresponsive before marking  it unhealthy. Must be N times more than kubelet's  nodeStatusUpdateFrequency, where N means number of retries allowed for  kubelet to post node status. 
- ```--pod-eviction-timeout duration```     Default: 5m0s | The grace period for deleting pods on failed nodes.

When the first threshold is met --> the virtual node becomes unhealty
When the second threshold is met --> the unjoin is performed